bitcoin/src/blockdata/script/witness_program.rs

```diff
@@ -27,43 +34,41 @@ pub struct WitnessProgram {
     /// The segwit version associated with this witness program.
     version: WitnessVersion,
     /// The witness program (between 2 and 40 bytes).
-    program: PushBytesBuf,
+    program: ArrayVec<u8, MAX_SIZE>,
 }
 
 impl WitnessProgram {
-    /// Creates a new witness program.
-    pub fn new<P>(version: WitnessVersion, program: P) -> Result<Self, Error>
-    where
-        P: TryInto<PushBytesBuf>,
-        <P as TryInto<PushBytesBuf>>::Error: PushBytesErrorReport,
-    {
+    /// Creates a new witness program, copying the content from the given byte slice.
+    pub fn new(version: WitnessVersion, bytes: &[u8]) -> Result<Self, Error> {
         use Error::*;
 
-        let program = program.try_into().map_err(|error| InvalidLength(error.input_len()))?;
-        if program.len() < 2 || program.len() > 40 {
-            return Err(InvalidLength(program.len()));
+        let program_len = bytes.len();
```

Comment:
Why not `program.len()`? I'm sure the method will be inlined so we don't have to worry about optimizations.

Reply:
FWIW I also baulked at the usage of `program_len` :)